### PR TITLE
CORE-12619: Also include custom AMQP serializers inside PERSISTENCE sandboxes.

### PR DESCRIPTION
**Pull request by chrisr3**
We need to be able to serialize / deserialize `ContractState` and `StateAndRef` in all sandbox group types, which means that their custom AMQP serializers also need to be available for `PERSISTENCE` sandboxes.

Custom serializers which do not use sandbox services can become "platform" services.

### CODE REVIEW

File: components/ledger/ledger-common-flow/src/main/kotlin/net/corda/ledger/common/flow/impl/transaction/filtered/serializer/amqp/FilteredTransactionSerializer.kt [hunk 1]
```diff
@@ -28,11 +28,15 @@ class FilteredTransactionSerializer @Activate constructor(
     private val merkleTreeProvider: MerkleTreeProvider
 ) : BaseProxySerializer<FilteredTransaction, FilteredTransactionProxy>(), UsedByFlow {
 
-    override val proxyType = FilteredTransactionProxy::class.java
+    override val proxyType
```
Comment (lankydan) [inline components/ledger/ledger-common-flow/src/main/kotlin/net/corda/ledger/common/flow/impl/transaction/filtered/serializer/amqp/FilteredTransactionSerializer.kt:31]: Whats the benefit of using `get()` here?

**Review by lankydan — COMMENTED** [review]
Change is fine left one comment.

File: components/ledger/ledger-common-flow/src/main/kotlin/net/corda/ledger/common/flow/impl/transaction/filtered/serializer/amqp/FilteredTransactionSerializer.kt (hunk 1 above)
Comment (chrisr3) [reply components/ledger/ledger-common-flow/src/main/kotlin/net/corda/ledger/common/flow/impl/transaction/filtered/serializer/amqp/FilteredTransactionSerializer.kt:31]: A `val` property like this:
```kotlin
override val proxyType = FilteredTransactionProxy::class.java
```
is compiled to something like this:
```java
private final Class<FilteredTransactionProxy> _proxyType = FilteredTransactionProxy.class;

public Class<FilteredTransactionProxy> getProxyType() {
    return _proxyType;
}
```
We don't really need that `_proxyType` field - all we need is the getter:
```java
public Class<FilteredTransactionProxy> getProxyType() {
    return FilteredTransactionProxy.class;
}
```

**Review by vlajos — APPROVED** [review]
LGTM

**Review by lankydan — APPROVED** [review]

**Review by wzur-r3 — APPROVED** [review]
LGTM

**Review by driessamyn — APPROVED** [review]
LGTM (for packaging)

**Pull request merged**